bin/weewx/reportengine.py

```diff
@@ -254,6 +254,113 @@ def _build_skin_dict(self, report):
+        # need to check both. config_dict (weewx.conf) has the final say,
+        # so check it first.
+        if 'lang' in self.config_dict['StdReport'][report]:
+            # 'lang' is set in weewx.conf
+            lang_config = "%s.conf" % self.config_dict['StdReport'][report]['lang']
+            log.debug("config_dict lang_config=%s for report %s" % (lang_config,report))
+        elif 'lang' in skin_dict:
+            # 'lang' is set in skin.conf
+            lang_config = "%s.conf" % skin_dict['lang']
+            log.debug("skin_dict lang_config=%s for report" % (lang_config,report))
+        else:
+            # No localization defined. Use defaults.
+            lang_config = None
+            log.debug("no language defined for report %s" % report)
+        
+        # If an localization file name could be determined, read the file.
+        if lang_config:
+        
+            # Now add the options in the report's localization file. Start by figuring where it is located.
+            lang_config_path = os.path.join(
+                self.config_dict['WEEWX_ROOT'],
+                self.config_dict['StdReport']['SKIN_ROOT'],
+                self.config_dict['StdReport'][report].get('skin', ''),
+                'lang',
+                lang_config)
+        
+            # Now retrieve the language dictionary for the skin. Wrap it in a try block in case we fail.  It is ok if
+            # there is no file - everything for a skin might be defined in the weewx configuration.
+            try:
+                merge_dict = configobj.ConfigObj(lang_config_path, file_error=True, encoding='utf-8')
+                log.debug("Found localization file %s for report '%s'", lang_config_path, report)
+                # make sure 'Texts' is present and a dict (section)
+                if not isinstance(merge_dict.get('Texts',None),dict):
+                    if 'Texts' in merge_dict:
+                        log.error("'Texts' is not a section in '%s'" % lang_config_path)
+                    merge_dict['Texts']={}
+                # set language code for $gettext.lang
+                merge_dict['Texts']['lang'] = os.path.basename(lang_config_path).split('.')[0]
+                # Merge the skin config file in:
+                weeutil.config.merge_config(skin_dict, merge_dict)
+                if self.first_run:
+                    log.info("Using localization file %s for report '%s'" % 
+                             (lang_config_path, report))
+            except IOError as e:
+                log.debug("Cannot read localization file %s for report '%s': %s",
+                          lang_config_path, report, e)
+            except SyntaxError as e:
+                log.error("Failed to read localization file %s for report '%s': %s",
+                          lang_config_path, report, e)
+                raise
+        
+        # The key 'unit_system' defines the unit system to be used with 
```

Comment:
Please. No `unit_system`. At least until we discuss how it interacts with the present system of unit groups.

Reply:
I guess that's the most difficult part. Let me try to explain. 

`defaults.py` is always `US`. That's the starting point. Imagine a user from a metric country. To get metric units in the report s/he has to copy all the long list of unit groups with their respective metric units to somewhere within `weewx.conf`. One by one. Line by line. A long, annoying and error-prone work.

And then, the user sees a wonderful, complete list of those unit groups together with their metric units in `units.py` (that is `weewx.units.MetricUnits`). But that list (or `dict` in Python) is not available to the user as a starting point when it comes to setting up reports.

Why that?

(And I changed the code for that heavily in the hope it would satisfy your needs then. It is very short now, and I used other names and hope they will be more descriptive.)

Reply:
OK, you're wearing me down. I just need to make sure that this will not break old skins in unexpected ways.